s3prl/downstream/a2a-vc-vctk/expert.py

```diff
@@ -0,0 +1,245 @@
+            shuffle=(sampler is None),
+            sampler=sampler,
+            num_workers=self.datarc['num_workers'],
+            collate_fn=dataset.collate_fn
+        )
+
+
+    def _get_eval_dataloader(self, dataset):
+        return DataLoader(
+            dataset, batch_size=self.datarc['eval_batch_size'],
+            shuffle=False, num_workers=self.datarc['num_workers'],
+            collate_fn=dataset.collate_fn
+        )
+
+
+    # Interface
+    def forward(self,
+                split,
+                input_features,
+                wavs,
+                acoustic_features,
+                acoustic_features_padded,
+                acoustic_feature_lengths,
+                wav_paths,
+                ref_spk_embs,
+                ref_spk_names, 
+                records,
+                **kwargs):
+
+        device = input_features[0].device
+
+        # padding
+        input_feature_lengths = torch.IntTensor([feature.shape[0] for feature in input_features])
+        input_features = pad_sequence(input_features, batch_first=True).to(device=device)
+        ref_spk_embs = ref_spk_embs.to(device)
+        
+        # forward model
+        if split in ["dev", "test"]:
+            predicted_features, predicted_feature_lengths = self.model(input_features, input_feature_lengths, ref_spk_embs)
+            # save the unnormalized features for dev and test sets
+            records["predicted_features"] += predicted_features.cpu().numpy().tolist()
+            records["feature_lengths"] += predicted_feature_lengths.cpu().numpy().tolist()
+            records["wav_paths"] += wav_paths
+            records["ref_spk_names"] += ref_spk_names
+            records["wavs"] += wavs
+        else:
+            predicted_features, predicted_feature_lengths = self.model(input_features, input_feature_lengths, ref_spk_embs, acoustic_features_padded.to(device))
+
+        # loss calculation (masking and normalization are done inside)
+        loss = self.objective(predicted_features,
+                              acoustic_features_padded,
+                              predicted_feature_lengths,
+                              acoustic_feature_lengths,
+                              device)
+        records['loss'].append(loss.item())
+
+        return loss
+
+
+    # interface
+    def log_records(self, split, records, logger, global_step, batch_ids, total_batch_num, **kwargs):
+        loss = torch.FloatTensor(records['loss']).mean().item()
+
+        if split in ["dev", "test"]:
+
+            hdf5_save_dir = os.path.join(self.expdir, str(global_step), split, "hdf5")
+            wav_save_dir = os.path.join(self.expdir, str(global_step), split, "wav")
+            os.makedirs(hdf5_save_dir, exist_ok=True)
+            os.makedirs(wav_save_dir, exist_ok=True)
+
+            for i, (wav_path, ref_spk_name) in enumerate(tqdm(zip(records["wav_paths"], records["ref_spk_names"]), dynamic_ncols=True, desc="Saving files")):
```

Comment:
Since the zip iterator does not expose the length to the tqdm bar, so tqdm cannot estimate the iteration/sec speech or the total execution time, making this tqdm less informative. Please add the `total` argument.

`tqdm(, total=len(records["wav_paths"]))`